### PR TITLE
ComboBox: Option rendering optimization

### PR DESCRIPTION
**Pull request by kysedate**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

The current ComboBox renders every option on every mouse/keyboard event. This is unnecessary and leads to very poor performance in ComboBoxes with many element. This change wraps the options in a ComboBoxOptionWrapper component, which compares props to see whether the comonent needs to be updated. The typical case is that options will perform an update when it receives or loses focus, but it also checks other props that would warrent a re-render.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 1]
```diff
@@ -77,6 +77,32 @@ enum HoverStatus {
...
+// This is used to customize when we want to rerender components,
+// so we don't rerender every option every time render is executed
+class ComboBoxOptionWrapper extends React.Component<IComboBoxOptionWrapperProps, {}> {
+  public render() {
+    return this.props.children;
+  }
+
+  public shouldComponentUpdate(newProps: IComboBoxOptionWrapperProps) {
+    for (const prop of Object.keys(newProps)) {
+      if (prop !== 'children' && this.props[prop] !== newProps[prop]) {
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Look at using shallowCompare in fabric Utilities for this comparison

**Comment by jspurlin** [comment]
@kysedate You need to bump the bundle size by .1kb

**Comment by jspurlin** [comment]
@dzearing you are listed as a required reviewer for this PR, we can't merge until you sign off (unless @manishgarg1 has override abilities)

**Comment by dzearing** [comment]
@kysedate If you pull master, the size has been updated and you should be able to merge.

**Pull request merged**